examples/distributed_bakery.rs

```diff
@@ -60,8 +61,8 @@ async fn increment_output(doc_handle: &DocHandle) -> Result<u32, ()> {
     Ok(latest)
 }
 
-async fn run_bakery_algorithm(doc_handle: &DocHandle, customer_id: &String) {
-    let (our_number, closing) = doc_handle.with_doc_mut(|doc| {
+async fn run_bakery_algorithm(doc_handle: &Mutex<DocHandle>, customer_id: &String) {
+    let (our_number, closing) = doc_handle.lock().await.with_doc_mut(|doc| {
```

Comment:
I think the locking behaviour here and elsewhere is probably wrong, but I'm not sure what to do...

@gterzian if you could advise?